Microsoft.Toolkit.Uwp.UI/Triggers/IsEqualStateTrigger.cs

```diff
@@ -61,12 +61,13 @@ public object To
 
         internal static bool AreValuesEqual(object value1, object value2, bool convertType)
         {
-            if (value1 == value2)
+            if (object.Equals(value1, value2))
             {
                 return true;
             }
 
-            if (value1 != null && value2 != null && convertType)
+            // If they are the same type but fail with Equals check, don't bother with conversion.
+            if (value1 != null && value2 != null && value1.GetType() != value2.GetType() && convertType)
```

Comment:
Would recommend to always use `is not null` when explicitly testing for `null`, to reduce the chance of errors. The two are effectively the same in this specific case, but we've started using `is not null` in the rest of the Toolkit lately for this reason, so it'd be nice to keep the consistent code style here as well 🙂

```suggestion
            if (value1 is not null && value2 is not null && value1.GetType() != value2.GetType() && convertType)
```